pgrx-tests/src/framework.rs

```diff
@@ -530,13 +597,27 @@ fn monitor_pg(mut command: Command, cmd_string: String, loglines: LogLines) -> S
         // exits. TODO: Consider finding a way to handle cases where we fail to
         // clean up due to a SIGNAL?
         add_shutdown_hook(move || unsafe {
-            libc::kill(pid as libc::pid_t, libc::SIGTERM);
-            let message_string = std::ffi::CString::new(
-                format!("stopping postgres (pid={pid})\n").bold().blue().to_string(),
-            )
-            .unwrap();
-            // IMPORTANT: Rust string literals are not naturally null-terminated
-            libc::printf("%s\0".as_ptr().cast(), message_string.as_ptr());
+            if let Some(_runas) = get_runas() {
+                Command::new("sudo")
+                    .arg("-u")
+                    .arg("root") // NB:  we must be "root" to kill the `sudo` process we spawned to start postgres
```

Comment:
Is there an alternative to using sudo to spawn it?

Reply:
Not that I can see.  The `sudo` process we're trying to kill is owned by root.

Reply:
Is there any way for us to, at this point, obtain intelligence on whether our problem was "the `kill` was ignored" or "`sudo` failed"?

I'm wondering if we should try to escalate to SIGKILL after. It seems... more important to consider now.

Reply:
All right then.